projectroles/utils.py

```diff
@@ -95,3 +110,192 @@ def get_app_names():
             else:
                 ret.append(s[0])
     return sorted(ret)
+
+
+class AppLinkContent:
+    """Class for generating application links for the UI"""
+
+    def _is_active_projectroles(self, request, link_names=None):
+        """Check if current URL is active under the projectroles app."""
+        # HACK: Avoid circular import
+        from projectroles.urls import urlpatterns
+
+        if request.resolver_match.app_name != 'projectroles':
+            return False
+        url_name = request.resolver_match.url_name
+        return url_name in [u.name for u in urlpatterns] and (
+            not link_names or url_name in link_names
+        )
+
+    def _is_active_plugin(self, app_plugin, app_name, url_name):
+        """
+        Check if current URL is active for a specific app plugin.
+        """
+        if app_plugin.name.startswith(app_name) and url_name in [
+            u.name for u in getattr(app_plugin, 'urls', [])
+        ]:
+            return True
+        # HACK for remote site views, see issue #1336
+        if (
+            app_name == 'projectroles'
+            and app_plugin.name == 'remotesites'
+            and url_name.startswith('remote_')
+        ):
+            return True
+        return False
+
+    def _is_app_visible(self, plugin, project, user):
+        """Check if app should be visible for user in a specific project."""
+        can_view_app = user.has_perm(plugin.app_permission, project)
+        app_hidden = False
+        if plugin.name in getattr(
+            settings, 'PROJECTROLES_HIDE_PROJECT_APPS', []
+        ):
+            app_hidden = True
+        if (
+            can_view_app
+            and not app_hidden
+            and (project.type == PROJECT_TYPE_PROJECT or plugin.category_enable)
+        ):
+            return True
+        return False
+
+    def _allow_project_creation(self):
+        """Check whether creating a project is allowed on the site."""
+        if (
+            settings.PROJECTROLES_SITE_MODE
+            == SODAR_CONSTANTS['SITE_MODE_TARGET']
+            and not settings.PROJECTROLES_TARGET_CREATE
+        ):
+            return False
+        return True
+
+    def get_app_links(self, request, project=None):
```

Comment:
Let's call this `get_project_app_links()` too, as we'll be adding a second method for the site apps for the user dropdown in the other issue.